src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Simple.cs

```diff
@@ -346,6 +346,14 @@ public void WaitForConnectionOnAlreadyConnectedServer_Throws_InvalidOperationExc
                 }
 
                 Assert.Throws<InvalidOperationException>(() => client.IsMessageComplete);
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    Assert.Throws<IOException>(() => client.NumberOfServerInstances);
+                }
+                else
+                {
+                    Assert.Throws<PlatformNotSupportedException>(() => client.NumberOfServerInstances);
```

Comment:
This line (355 in NamedPipeTest.Simple.cs) is throwing an "IOException: Pipe is broken" on Linux for:

- Server: InOut, Client: Out
- Server: In, Client: Out